### PR TITLE
[rc] Add variant for the latest AWS SDK on the `libevent-2.1.10` spec.

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

See added comment in `recipe/conda_build_config.yaml`

### CODE REVIEW

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Review by DrNickClarke — APPROVED** [review]

**Comment by jjerphan** [comment]
Failures are due to crash in a test suite.

Likely related to new minor release of libraries in the AWS stack.

Unlikely related to the update of the C standard library (from 2.16 to 2.17).

Diff of dependencies between [the previous build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=967917&view=logs&j=4b3757f9-99c5-5891-6c0c-93d7030cdf7d&t=f047df50-a233-5ae9-bf8f-c7dd8e5684ad) and [this one](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=974800&view=logs&j=4b3757f9-99c5-5891-6c0c-93d7030cdf7d&t=f047df50-a233-5ae9-bf8f-c7dd8e5684ad&l=2213) for `osx_64_aws_sdk_cpp_azure_core_cpp_libevent_numpy1.22python3.9.____cpython`:
```diff
1,2c1,2
< aws-c-auth:               0.7.22-h2f80047_6           conda-forge
< aws-c-cal:                0.6.15-hd73d8db_1           conda-forge
---
> aws-c-auth:               0.7.22-he52cbe7_7           conda-forge
> aws-c-cal:                0.7.0-hd73d8db_0            conda-forge
6,7c6,7
< aws-c-http:               0.8.2-hc7634fe_3            conda-forge
< aws-c-io:                 0.14.9-h2089a17_3           conda-forge
---
> aws-c-http:               0.8.2-hf609411_4            conda-forge
> aws-c-io:                 0.14.9-h9f49e27_5           conda-forge
9c9
< aws-c-s3:                 0.5.10-h7347a4b_4           conda-forge
---
> aws-c-s3:                 0.6.0-h8fdb93b_0            conda-forge
12,13c12,13
< aws-crt-cpp:              0.26.12-hc167df4_1          conda-forge
< aws-sdk-cpp:              1.11.329-h4fac305_6         conda-forge
---
> aws-crt-cpp:              0.27.2-h7746516_0           conda-forge
> aws-sdk-cpp:              1.11.329-hce9e41e_8         conda-forge
22c22
< ca-certificates:          2024.6.2-h8857fd0_0         conda-forge
---
> ca-certificates:          2024.7.4-h8857fd0_0         conda-forge
42c42
< libcxx:                   17.0.6-h8d2f0da_2           conda-forge
---
> libcxx:                   18.1.8-hef8daea_0           conda-forge
56c56
< libopenblas:              0.3.27-openmp_hfef2a42_0    conda-forge
---
> libopenblas:              0.3.27-openmp_h8869122_1    conda-forge
83,84c83,84
< setuptools:               70.1.1-pyhd8ed1ab_0         conda-forge
< snappy:                   1.2.0-h6dc393e_1            conda-forge
---
> setuptools:               70.2.0-pyhd8ed1ab_0         conda-forge
> snappy:                   1.2.1-he1e6707_0            conda-forge
```

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -14,29 +14,37 @@ xxhash:
...
 # Coupling dependencies specification together.
 # The first  line comes from the specifications of the `libevent-2.1.10` branch.
-# The second line comes from the alternative specifications of the `libevent-2.1.10` branch.
-# The third  line comes from the specifications of the `main` branch.
+# The second and third line come from the alternative specifications of the `libevent-2.1.10` branch.
+# The fourth line comes from the specifications of the `main` branch.
 # See: https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#coupling-keys
+# "*" means that the dependency is chosen implicitly to be the one of
+# conda-forge global pinning:
+# See: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:22]: This is not what `"*"` means, unfortunately; what this does is _override_ the global pinning with "any version goes" constraint, and so you will always pull in the newest version, even if conda-forge hasn't migrated for this version yet.

Comment (jjerphan) [reply recipe/conda_build_config.yaml:22]: Thank you for pointing this out.

I forgot that the versions on the global pinning were not necessarily the last ones. We wanted to have the latest versions for the AWS SDK, hence only the comment need to be changed.

Comment (jjerphan) [reply recipe/conda_build_config.yaml:22]: Corrected with 817366cfef031f32a3a95148cf5d52b1a1b473bc (a `[ci skip]` commit pushed to `rc` directly).

**Pull request merged**